packages/lpa-questionnaire-web-app/src/controllers/upload-plans.js

```diff
@@ -8,3 +9,80 @@ exports.getUploadPlans = (req, res) => {
     uploadedFiles: [],
   });
 };
+
+exports.postUploadPlans = async (req, res) => {
+  const documents = req.body?.files?.documents || [];
+
+  const { delete: deleteFile = '', errors = {}, submit = '' } = req.body;
+
+  // Chance for delete to be triggered due to non-JS solution. delete will be set to value of filename if button clicked
+  if (deleteFile) {
+    // TODO: add handling for deletion from DB and doc store as part of AS-1538
+    const file = req.session.uploadedFiles?.find((upload) => upload.name === deleteFile);
+
+    if (file) {
+      req.session.uploadedFiles = req.session.uploadedFiles.filter(
+        (upload) => upload.name !== deleteFile
+      );
+    }
+  } else if (Array.isArray(documents) && documents.length) {
```

Comment:
Looks like `Array.isArray(documents)` would always be true due to line 14?

Reply:
Good catch, line 14 change happened later than this line